cwe_checker_rs/src/intermediate_representation/term.rs

```diff
@@ -68,6 +68,33 @@ pub enum Def {
     Assign { var: Variable, value: Expression },
 }
 
+impl Term<Def> {
+    /// This function checks whether the following instruction
+    /// is a zero extension of the currently overwritten sub register and
+    /// if so, returns its TID
+    pub fn check_for_zero_extension(&self, output_name: String) -> Option<Tid> {
+        match &self.term {
+            Def::Assign { var, value } => {
+                if output_name == var.name {
```

Comment:
I think you can combine that to `Def::Assign {var, value } if output_name == var.name => {` to get rid of a nesting level here.